lxd/storage_pools_utils.go

```diff
@@ -183,3 +186,30 @@ func dbStoragePoolDeleteAndUpdateCache(s *state.State, poolName string) error {
 
 	return err
 }
+
+// filterUsedBy filters a UsedBy list based on project access
+func filterUsedBy(d *Daemon, r *http.Request, entries []string) []string {
+	// Shortcut for admins and non-RBAC environments.
+	if d.userIsAdmin(r) {
+		return entries
+	}
+
+	// Filter the entries.
+	usedBy := []string{}
+	for _, entry := range entries {
+		projectName := project.Default
+		fields := strings.Split(entry, "?project=")
```

Comment:
@stgraber this could be rather brittle in case the generated URLs have `project=` as a non-first argument in the query string. In the future we may add an additional unrelated argument and the developer may put that one first in the generated output not knowing the security impact it would have (as query string parsing isn't usually position dependent).

How about using https://golang.org/pkg/net/url/#ParseQuery instead to make this more resilience to different ordered query strings?

Reply:
Yeah, I'll do that when I generalize it. For the current use case I had validated that project was always first :)

Reply:
Yeah true, so wouldn't have leaked info, just would be missing some entries.